docs/design/kademlia-removal.md

```diff
@@ -0,0 +1,84 @@
+# Kademlia Removal
+
+## Abstract
+
+This design document outlines the communication protocol between satellites and 
+storage nodes, network refreshes, and kademlia removal given the satellite opt-in 
+capability for storage nodes. 
+
+## Background
+User research on how Storage Node Operators would connect to satellites resulted in a 
+simple solution called Opt-in SNO select. Satellites wait for storage nodes to work 
+with them, and storage nodes manually select the satellites they want to work with. 
+The initial implementation of this allows SNOs to update their trusted satellite list 
+in a hardcoded configuration file, but future improvements will enable users to manage 
+this list through a web console. Due to the nature of this change, the Storj network 
+at this point in time no longer requires the Kademlia DHT or related entities. 
+We will replace this protocol with direct communication between satellites and 
+storage nodes, and keep the network fresh without kademlia node discovery and random 
+lookups.
+
+## Design
+
+1. Nodes reach out to satellites that they want to work with
+    
+    - At the time of writing, a storage node operator can input a list of trusted satellite 
+    IDs and addresses into their configuration file on setup. There are several tardigrade-level 
+    satellites already listed as a preset. This list is imported into a Go map data structure 
+    within the “trust” package.
+    - Eventually during the design and implementation of trusted satellite management, 
+    this configured list may be imported into a on-disk data store, eg SQL table. Users will 
+    be able to modify this list through the storage node operator console. However, this is a 
+    separate discussion from this design document. This design is satisfied with a static trusted 
+    satellites list.
+    - Nodes will be able to communicate with satellites through the “transport” and “trust” packages 
+    directly rather than using kademlia to traverse the network to find the address of a given ID.
```

Comment:
It might be worth having a section on NodeIDs themselves.  Without Kad, the value proposition of NodeIDs changes.

Reply:
Could you elaborate on what you mean here @wthorp ? I suppose we may not need the node dossier any longer either since we won't be exposing node info to other nodes on the network

Reply:
i think he's saying we should reconsider if everything about node ids is still necessary (such as difficulty, construction based on certificate authority public key, etc).

that said, i think that's worth another design document. i also don't thiiiiink we'll want to make any node id changes anyways. the difficulty and the construction are still useful for other reasons.

Reply:
sounds good! I will make some notes about subsequent design docs

Reply:
> i also don't thiiiiink we'll want to make any node id changes anyways

Whether we make changes or not, having a degree of difficulty in generating a valid nodeID is a useful feature, as it prevents users from flooding the network with bogus nodes just for the lulz. There are other ways (other than "proof of work") that this could be done, but POW is likely the easiest decentralized way to gatekeep noes coming onto the network.